map/src/infoblock/components/graph/GpxGraphProvider.jsx

```diff
@@ -222,8 +223,20 @@ const GpxGraphProvider = ({ width }) => {
                 calculatedSlope[k] = 0;
             }
         }
-        let res = [];
+        // add slopes to points
+        for (let i = 0; i < result.length; i++) {
+            let current = result[i];
+            if (current) {
+                let dist = current[DISTANCE];
+                let ind = calculatedSlopeDist.findIndex((d) => d / 1000 > dist - 0.005 && d / 1000 < dist + 0.005);
+                if (ind) {
+                    result[i][SLOPE] = Math.trunc(calculatedSlope[ind] * 100) / 100;
+                }
```

Comment:
double recheck if()

findIndex returns -1 if not found (and might return 0 - 1st element of array)